Objects/unicodeobject.c

```diff
@@ -14617,6 +14617,115 @@ unicode_new_impl(PyTypeObject *type, PyObject *x, const char *encoding,
     return unicode;
 }
 
+static const char *
+as_const_char(PyObject *obj, const char *name)
+{
+    if (!PyUnicode_Check(obj)) {
+        PyErr_Format(PyExc_TypeError,
+                     "str() argument '%s' must be str, not %T",
+                     name, obj);
+        return NULL;
+    }
+    const char *str = _PyUnicode_AsUTF8NoNUL(obj);
+    if (str == NULL) {
+        return NULL;
+    }
+    return str;
+}
+
+static PyObject *
+fallback_to_tp_call(PyObject *type, Py_ssize_t nargs, Py_ssize_t nkwargs,
```

Comment:
Would it be possible to inline this function in unicode_vectorcall(), using goto if needed? The function name is too generic, it doesn't mention unicode_new().

Reply:
Sure; that was my original code, but I refactored it out in order to reduce the size of the vectorcall function.

Reply:
I think the code is nicer with this refactoring in place though; perhaps a better name would suffice.